File: androidApp/src/main/java/com/andremw96/notesnotes_kmm/android/ui/addeditnote/AddEditNoteState.kt
package com.andremw96.notesnotes_kmm.android.ui.addeditnote

/**
 * Data validation state of the login form.
 */
data class AddEditNoteState(
    val noteId: Int = -1,
    val title: String = "",
    val description: String = "",
    val titleError: String? = null,
    val descriptionError: String? = null,
    val isLoading: Boolean = false,
    val isNewNote: Boolean = false,
    val saveNoteSuccess: Boolean = false,
    val saveNoteError: String? = null,
) {
    val isDataValid: Boolean
        get() = titleError == null && descriptionError == null && title.isNotEmpty() && description.isNotEmpty()
}

File: iosApp/iosApp/Screens/AddEditNote/AddEditNoteState.swift
//
//  AddEditNoteState.swift
//  iosApp
//
//  Created by Quipper Indonesia on 13/01/23.
//  Copyright © 2023 orgName. All rights reserved.
//

import Foundation

struct AddEditNoteState {
    var isLoading = false
    var noteItem: NoteItem = NoteItem(createdAt: "", description: "", id: -1, isDeleted: false, title: "", updatedAt: "", userId: -1)
    var titleError = ""
    var descriptionError = ""
    var isEditing = false
    var saveNoteSuccess: Bool? = nil
    var saveNoteError = ""
    var authenticationError = ""
    
    var isDataValid: Bool {
        get { return titleError == "" && descriptionError == "" && saveNoteError == "" && noteItem.title != "" && noteItem.description != "" }
    }
    
    var isSaveError: Bool {
        get { return saveNoteError != "" || (saveNoteSuccess != nil && saveNoteSuccess == false) }
    }
    
    var isAuthError: Bool {
        get { return authenticationError != "" }
    }
}
